catalog/plugin/catalog-plugin-resource-usage/src/test/java/ddf/catalog/plugin/resource/usage/TestResourceUsagePlugin.java

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin.resource.usage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.io.Serializable;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.shiro.authz.Permission;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.mgt.SecurityManager;
+import org.apache.shiro.realm.AuthorizingRealm;
+import org.apache.shiro.realm.Realm;
+import org.apache.shiro.session.mgt.SimpleSession;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.subject.support.DelegatingSubject;
+import org.codice.ddf.persistence.attributes.AttributesStore;
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+
+public class TestResourceUsagePlugin {
+
+    ResourceUsagePlugin plugin;
+
+    AttributesStore attributeStore;
+
+    AttributesStore emptyAttributeStore;
+
+    private static final String TEST_USER = "testuser";
+
+    private static final String RESOURCE_SIZE = "100";
+
+    Subject subject;
+
+    @Before
+    public void setup() {
+        attributeStore = spy(new AttributesStore() {
```

Comment:
It seems odd to perform assertions on the method arguments inside a `spy`. Isn't that what `ArgumentCaptor`s are for? If you went that route, you could use a `mock` instead of a `spy`.